### ISSUE TITLE
Build require `rsvg-convert` and `imagick` even with `-with-gui=no`

### ISSUE THREAD

**Issue by Retropex** [bitcoinknots/bitcoin#79]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

Fail to generate the `Makefile`

### Expected behaviour

Be able to build without `rsvg-convert` and `imagick` when the argument `-with-gui=no` is passed.

### Steps to reproduce

``` bash
./autogen
./configure --with-gui=no
make
```

### How did you obtain Bitcoin Knots

Compiled from source

### What version of Bitcoin Knots are you using?

v26.1

### Operating system and version

Debian 12.5 and macOS Sonoma 14.4.1

**Comment by luke-jr** [comment]
Should be fixed in 26.1.knots20240513, please confirm

**Comment by Retropex** [comment]
tested, and it's not working

logs:

```
checking whether C++ compiler accepts -march=armv8-a+crc+crypto... no
checking whether C++ compiler accepts -march=armv8-a+crypto... no
checking for ARMv8 CRC32 intrinsics... no
checking for ARMv8 SHA-NI intrinsics... no
checking for POWER8 compiler support... no
checking for rsvg-convert... no
checking for rsvg... no
checking for convert... no
checking for png2icns... no
configure: error: Couldn't find rsvg-convert (required to build GUI from git)
```

tested on fresh VM with this package installed:

```
sudo apt update && sudo apt-get install -y build-essential libtool autotools-dev automake pkg-config bsdmainutils python3 && sudo apt-get install -y libevent-dev libboost-dev && sudo apt install -y libsqlite3-dev && sudo apt install -y libminiupnpc-dev libnatpmp-dev && sudo apt-get install -y libzmq3-dev && sudo apt install -y systemtap-sdt-dev
```

**Comment by renemadeira** [comment]
Although is already confirmed, I can attest the bug. All the dependencies for the GUI must be installed.
It is not recognizing at all the argument, I tried "--without-gui" and "--with-gui=0".
When installing the dependencies, the generated makefile will carry the option "with gui / qt   = yes".

**Comment by luke-jr** [comment]
The issue with the fix is that it is checking *before* we know if the GUI is wanted. Maybe that check can be simply moved later - anyone want to try?

**Comment by renemadeira** [comment]
I want!
I'll be back in the computer in half an hour.

**Comment by renemadeira** [comment]
Can you point me where is this check happening?

**Comment by luke-jr** [comment]
Most of the individual program checks are around line 735. There's a macOS-specific one at line 807. The decision to throw an error (this is probably the part that should move) is around line 915. `bitcoin_enable_qt` is defined at line 1558.

**Comment by luke-jr** [comment]
After modifying `configure.ac`, you will need to run `autogen.sh` again manually

**Comment by renemadeira** [comment]
Okay, I noticed some weird things.
rsgv-convert and imagemagick are tested in lines 735 to 744, but there is no check of bitcoin_enable_qt variable at all to perform those checks.
Besides that, the --with-gui=no build doesn't work without qt dependencies, too.
When installing the dependencies just to get through the checks, the configure process will finish with the gui enabled.
However, if I pass --without-gui argument, the configure process requires the same dependencies, but will end up with the gui disabled.

**Comment by luke-jr** [comment]
There is no `--disable-gui` option...

**Comment by renemadeira** [comment]
> There is no `--disable-gui` option...

Sorry, I wrote all wrong. Corrected now.

**Comment by luke-jr** [comment]
Please confirm this is fixed in 27.1.knots20260621 and close if so

**Comment by Retropex** [comment]
it work's on Debian Bookworm
